engine/execution/computation/computer/result_collector.go

```diff
@@ -171,13 +190,32 @@ func (collector *resultCollector) CommitCollection(
 
 func (collector *resultCollector) Stop() {
 	collector.closeOnce.Do(func() {
-		close(collector.viewChan)
-		close(collector.eventsListChan)
+		close(collector.committerInputChan)
+		close(collector.hasherInputChan)
 	})
 }
 
-func (collector *resultCollector) Finalize() *execution.ComputationResult {
+func (collector *resultCollector) Finalize() (
+	*execution.ComputationResult,
+	error,
+) {
 	collector.Stop()
-	collector.wg.Wait()
-	return collector.result
+
+	<-collector.committerDoneChan
+	<-collector.hasherDoneChan
+
+	var err error
+	if collector.committerError != nil {
+		err = multierror.Append(err, collector.committerError)
+	}
+
+	if collector.hasherError != nil {
+		err = multierror.Append(err, collector.hasherError)
+	}
```

Comment:
nit: you can likely simplify this as:
```suggestion
	multierr.AppendInto(&err, collector.committerError)
	multierr.AppendInto(&err, collector.hasherError)
```

Reply:
there's no such function? https://pkg.go.dev/github.com/hashicorp/go-multierror#pkg-index